micromagneticdata/drive.py

```diff
@@ -415,3 +416,25 @@ def slider(self, description="step", **kwargs):
         return ipywidgets.IntSlider(
             value=0, min=0, max=self.n - 1, step=1, description=description, **kwargs
         )
+
+    def to_xarray(self):
+        """Export ``micromagneticdata.Drive`` as ``xarray.Dataset``"""
+        if self.info["driver"] == "MinDriver":
+            darray = self[0].to_xarray()
+        else:
+            field_darrays = (self[i].to_xarray() for i in range(self.n))
+            if self.info["driver"] == "TimeDriver":
+                darray = xr.concat(field_darrays, dim="t").assign_coords(
+                    {"t": self.table.data["t"].to_numpy()}
+                )
+            else:
+                darray = xr.concat(field_darrays, dim="B_hysteresis").assign_coords(
+                    {
+                        f"B{i}": ("B_hysteresis", self.table.data[f"B{i}_hysteresis"])
+                        for i in "xyz"
+                    }
+                )
+
+        return xr.Dataset({"fields": darray, "m0": self.m0.to_xarray()}).assign_attrs(
+            **self.info
+        )
```

Comment:
For time-series it could sometimes be more convenient if `m0` would be stored as `fields[0]` and the first timestep (`drive[0]`) was `fields[1]` (and the whole object could then be a `DataArray`).

Downside: this breaks all our current conventions and might easily lead to confusion.

If we would think about such functionality the user interface could e.g. look like
```python
Drive.to_xarray(self, separate_m0=True)
```

If the user then passes `separate_m0=False` a single `DataArray` with `m0` being the first "part" is returned.

Any thoughts @marijanbeg and @fangohr ?

Reply:
I agree the functionality would be good to have, and that this can be confusing. 

I like the idea of an extra keyword argument. 

How about `prefix_m0` instead of `separate_m0`? The idea is that we put `m0` in front of all the other snapshots of `m` (either as a *prefix* (noun) or by *prefixing* m0 (as a verb)).

Reply:
I like the idea of passing an argument to store `m0` as `fields[0]`. I think by default we should keep `fields` and `m0` separate to be consistent with `Drive`. Also, to make it less confusing (and consistent) for the user when prefixing `m0`, we can still return an `xr.Dataset` with just one `xr.DataArray` named `fields`. Since I am thinking of keeping the two separate, I am inclined to use `prefix_m0`. What do you think @lang-m, @fangohr and @marijanbeg ?

Reply:
I'm happy to go with `prefix_m0`.

I am not so sure about the return value. `xr.Dataset` would maybe be more consistent, `xr.DataArray` more useful (in most situations) I guess.